src/java.base/share/classes/javax/crypto/KDF.java

```diff
@@ -0,0 +1,688 @@
+     * the preferred provider order for the specified algorithm. This
+     * may be different than the order of providers returned by
+     * {@link Security#getProviders() Security.getProviders()}.
+     * See also the Cipher Transformations section of the {@extLink
+     * security_guide_jdk_providers JDK Providers} document for information
+     * on the transformation defaults used by JDK providers.
+     *
+     * @throws NoSuchAlgorithmException
+     *     if no {@code Provider} supports a {@code KDF} implementation for the
+     *     specified algorithm
+     * @throws NullPointerException
+     *     if {@code algorithm} is {@code null}
+     */
+    public static KDF getInstance(String algorithm)
+        throws NoSuchAlgorithmException {
+        Objects.requireNonNull(algorithm, "algorithm must not be null");
+        try {
+            return getInstance(algorithm, (KDFParameters) null);
+        } catch (InvalidAlgorithmParameterException e) {
+            throw new NoSuchAlgorithmException(
+                "No implementation found using null KDFParameters", e);
+        }
+    }
+
+    /**
+     * Returns a {@code KDF} object that implements the specified algorithm from
+     * the specified security provider. The specified provider must be
+     * registered in the security provider list.
+     *
+     * @param algorithm
+     *     the key derivation algorithm to use.
+     *     See the {@code KDF} section in the <a href=
+     *     "{@docRoot}/../specs/security/standard-names.html#kdf-algorithms">
+     *     Java Security Standard Algorithm Names Specification</a>
+     *     for information about standard KDF algorithm names.
+     * @param provider
+     *     the provider to use for this key derivation
+     *
+     * @return a {@code KDF} object
+     *
+     * @throws NoSuchAlgorithmException
+     *     if the specified provider does not support the specified {@code KDF}
+     *     algorithm
+     * @throws NoSuchProviderException
+     *     if the specified provider is not registered in the security provider
+     *     list
+     * @throws NullPointerException
+     *     if the {@code algorithm} or {@code provider} is {@code null}
+     */
+    public static KDF getInstance(String algorithm, String provider)
+        throws NoSuchAlgorithmException, NoSuchProviderException {
+        Objects.requireNonNull(algorithm, "algorithm must not be null");
+        Objects.requireNonNull(provider, "provider must not be null");
+        try {
+            return getInstance(algorithm, null, provider);
+        } catch (InvalidAlgorithmParameterException e) {
+            throw new NoSuchAlgorithmException(
+                "No implementation found using null KDFParameters", e);
+        }
+    }
+
+    /**
+     * Returns a {@code KDF} object that implements the specified algorithm from
+     * the specified security provider. The specified provider must be
+     * registered in the security provider list.
```

Comment:
The second sentence should be removed. The Provider is passed in so does not need to be registered.